pyvista/core/datasetattributes.py

```diff
@@ -750,8 +749,24 @@ def _prepare_array(
         """Prepare an array to be added to this dataset."""
         if data is None:
             raise TypeError('``data`` cannot be None.')
-        if isinstance(data, Iterable):
-            data = pyvista_ndarray(data)
+
+        # attempt to reuse the existing pointer to underlying VTK data
+        if isinstance(data, pyvista_ndarray):
+            # pyvista_ndarray already contains the reference to the vtk object
+            # pyvista needs to use the copy of this object rather than wrapping
+            # the array (which leaves a C++ pointer uncollected.
+            if data.VTKObject is not None:
+                # VTK doesn't support strides, therefore we can't directly
+                # point to the underlying object
+                if data.__array_interface__['strides'] is None:
+                    vtk_arr = copy_vtk_array(data.VTKObject, deep=deep_copy)
+                    if isinstance(name, str):
+                        vtk_arr.SetName(name)
+                        return vtk_arr
+
+        # convert to numpy type
+        if not isinstance(data, np.ndarray):
+            data = np.asarray(data)
```

Comment:
I think this could just be
```suggestion
        # convert to numpy type if necessary
        data = np.asarray(data)
```
because if the data is already an ndarray then `np.asarray(data) is data`.